connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigBackingStore.java

```diff
@@ -711,9 +742,9 @@ KafkaBasedLog<String, byte[]> setupAndCreateKafkaBasedLog(String topic, final Wo
         return createKafkaBasedLog(topic, producerProps, consumerProps, new ConsumeCallback(), topicDescription, adminSupplier);
     }
 
-    private void sendPrivileged(String key, byte[] value) {
+    private void sendPrivileged(String key, byte[] value) throws ExecutionException, InterruptedException {
         if (!usesFencableWriter) {
-            configLog.send(key, value);
+            configLog.send(key, value).get();
```

Comment:
Unbounded waiting for the record to send doesn't seem very safe here. It looks like we were already trying to achieve synchronous writes by immediately performing a read-to-end after the call to `configLog::send` with a timeout of `READ_TO_END_TIMEOUT_MS`; could we use that same timeout while waiting for producer send futures to be completed (possibly after renaming to reflect the new purpose of the timeout)?

Reply:
Also, making this synchronously await the producer future's completion has the potential to slow things down for operations that involve writing multiple records to the config topic (such as [deleting a connector](https://github.com/apache/kafka/blob/b2cb546fba03bbdc4054c7d120b0b2654c7cf34e/connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigBackingStore.java#L493-L494)).

This is already the case with exactly-once support enabled since we perform each write in its own transaction, but while we're in the neighborhood, if there's an easy way to group together producer sends before awaiting the completion of any of their futures, it'd be nice. Not going to block on this, though.